app/models/post.rb

```diff
@@ -1,3 +1,15 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  after_create :update_posts_counter_user
+
+  def update_posts_counter_user
+    author.update(post_counter: author.posts.count)
+  end
```

Comment:
## Optional
- [ ] I suggest that you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the `#update_posts_counter_user` method every time a new post has been saved to the database, then make the `#update_posts_counter_user` method private